CMakeLists.txt

```diff
@@ -87,6 +87,14 @@ elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "s390x")
     if(${OQS_DIST_BUILD})
         set(OQS_DIST_S390X_BUILD ON)
     endif()
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "riscv")
+    if(BOARD MATCHES "qemu_riscv32")
```

Comment:
`BOARD` is a Zephyr variable. I‘m unsure if we should consider Zephyr specific stuff in the general CMakeLists.txt (as some users may want to build for riscv outside of the Zephyr port.
In my opinion it would be better to just decide on the 32/64bit Arch setting based on the `CMAKE_SYSTEM_PROCESSOR` variable (which you already prepared properly in the `Zephyr/CMakeLists.txt`.

Reply:
Haha no :D tyour feedback has been perfect :) 

The comment relates to the changes introduced by this PR (not mine). In my initial PR, I tried to only use Zephyr CMake variables within the `zephyr/CMakeLists.txt` file and prepare everything arch related for the general `CMakeLists.txt`, which will then ultimately be used. 
In this PR, @trigpolynom extended that preparation in the `zephyr/CMakeLists.txt` file (see below, where `CMAKE_SYSTEM_PROCESSOR` is set), but then again uses the Zephyr variables, not the prepared general variables in the main `CMakeLists.txt` file. Instead, I would suggest using the prepared variables here to not have any Zephyr reference in the main `CMakeLists.txt`.

Reply:
removed any board references in the root CMakeLists.txt!